src/EditorFeatures/Core/Tagging/TaggerContext.cs

```diff
@@ -22,7 +22,7 @@ internal class TaggerContext<TTag> where TTag : ITag
     {
         private readonly ImmutableDictionary<ITextBuffer, TagSpanIntervalTree<TTag>> _existingTags;
 
-        internal IEnumerable<DocumentSnapshotSpan> _spansTagged;
+        internal ImmutableArray<SnapshotSpan> _spansTagged;
```

Comment:
_spansTagged never made use of the 'Document' portoin of hte span.  And the Document part was scary (to me) as we mutate the document when computing these (e.g. we get teh FrozenPartial snapshot.  i really didn't like accidentally having that live and be out of sync with the actual Doc being classified).